.github/actions/build-module/action.yml

```diff
@@ -0,0 +1,44 @@
+name: 'Build Module'
+description: 'Builds the viamrtsp module using Docker'
+inputs:
+  target_os:
+    description: 'Target OS'
+    required: true
+  target_arch:
+    description: 'Target architecture'
+    required: true
+  docker_image:
+    description: 'Docker image to use for building'
+    required: true
+runs:
+  using: 'composite'
+  steps:
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.21.13
+
+    - name: Pull Docker image
+      run: docker pull ${{ inputs.docker_image }}
```

Comment:
so using with: and including a docker image in the other workflows makes that an input here? i'm not familiar with inputs

Reply:
yeah I think of it as params and args

params being what's defined above in `inputs:`

and `with:` being the args passed along